src/processor/mapper/map_aggregate.cpp

```diff
@@ -14,56 +14,80 @@ using namespace kuzu::planner;
 namespace kuzu {
 namespace processor {
 
+static std::vector<std::unique_ptr<AggregateInputInfo>> getAggregateInputInfos(
+    const expression_vector& groupByExpressions, const expression_vector& aggregateExpressions,
+    const Schema& schema) {
+    // Collect unFlat groups from
+    std::unordered_set<f_group_pos> groupByGroupPosSet;
+    for (auto& expression : groupByExpressions) {
+        groupByGroupPosSet.insert(schema.getGroupPos(*expression));
+    }
+    std::unordered_set<f_group_pos> unFlatAggregateGroupPosSet;
+    for (auto groupPos : schema.getGroupsPosInScope()) {
+        if (groupByGroupPosSet.contains(groupPos)) {
+            continue;
+        }
+        if (schema.getGroup(groupPos)->isFlat()) {
+            continue;
+        }
+        unFlatAggregateGroupPosSet.insert(groupPos);
+    }
+    std::vector<std::unique_ptr<AggregateInputInfo>> result;
+    for (auto& expression : aggregateExpressions) {
+        DataPos aggregateVectorPos{};
+        if (expression->getNumChildren() != 0) { // COUNT(*) has no children
+            auto child = expression->getChild(0);
+            aggregateVectorPos = DataPos{schema.getExpressionPos(*child)};
+        }
+        std::vector<data_chunk_pos_t> multiplicityChunksPos;
+        for (auto& groupPos : unFlatAggregateGroupPosSet) {
+            if (groupPos != aggregateVectorPos.dataChunkPos) {
```

Comment:
@ray6080 mentioned last time that: it is better to reserve enough space for valueVector before the `for` loop. So we can avoid dynamically resize the vector inside the for loop.